pkg/scheduler/scheduler.go

```diff
@@ -489,9 +489,12 @@ func (s *Scheduler) getAssignments(log logr.Logger, wl *workload.Info, snap *cac
 	if wl.CanBePartiallyAdmitted() {
 		reducer := flavorassigner.NewPodSetReducer(wl.Obj.Spec.PodSets, func(nextCounts []int32) (*partialAssignment, bool) {
 			assignment := flvAssigner.Assign(log, nextCounts)
-			if assignment.RepresentativeMode() == flavorassigner.Fit {
```

Comment:
Please reverted formatting changes if not needed (to miminize distractions).

Reply:
there is no reason the call `RepresentativeMode` twice

Reply:
I would suggest opening a dedicated cleanup PR, doesn't see to have anything to do with the fix.

Reply:
done #3078